### ISSUE TITLE
Increasing loss

### ISSUE THREAD

**Issue by anadb** [Xilinx/brevitas#118]
Hello I wrote a simple script to test MNIST data

```
train = datasets.MNIST('', train=True, download=True,
                       transform=transforms.Compose([
                           transforms.ToTensor()
                       ]))

test = datasets.MNIST('', train=False, download=True,
                       transform=transforms.Compose([
                           transforms.ToTensor()
                       ]))


trainset = torch.utils.data.DataLoader(train, batch_size=10, shuffle=True)
testset = torch.utils.data.DataLoader(test, batch_size=10, shuffle=False)

class QuantNet(nn.Module):
    def __init__(self):
        super().__init__()
        self.fc1   = qnn.QuantLinear(28*28, 64, bias=True, 
                                     weight_quant_type=QuantType.INT, 
                                     weight_bit_width=4)
        self.relu1 = qnn.QuantReLU(quant_type=QuantType.INT, bit_width=8, max_val=6)
        self.fc2   = qnn.QuantLinear(64, 64, bias=True, 
                                     weight_quant_type=QuantType.INT, 
                                     weight_bit_width=4)
        self.relu2 = qnn.QuantReLU(quant_type=QuantType.INT, bit_width=8, max_val=6)
        self.fc3   = qnn.QuantLinear(64, 64, bias=True, 
                                     weight_quant_type=QuantType.INT, 
                                     weight_bit_width=4)
        self.relu3 = qnn.QuantReLU(quant_type=QuantType.INT, bit_width=8, max_val=6)
        self.fc4   = qnn.QuantLinear(64, 10, bias=True, 
                                     weight_quant_type=QuantType.INT, 
                                     weight_bit_width=4)

    def forward(self, x):
        x = self.relu1(self.fc1(x))
        x = self.relu2(self.fc2(x))
        x = self.relu3(self.fc3(x))
        x = self.fc4(x)
        return F.log_softmax(x, dim=1)

qnet = QuantNet()

optimizer = optim.Adam(qnet.parameters(), lr=0.001)

for epoch in range(3):
    for data in trainset:
        X, y = data
        qnet.zero_grad()
        output = qnet(X.view(-1,784))
        loss = F.nll_loss(output, y)
        loss.backward()
        optimizer.step()
    print(loss)
```
During the training the loss is not continuously decreasing, what could it be?

```
tensor(0.0672, grad_fn=<NllLossBackward>)
tensor(0.0136, grad_fn=<NllLossBackward>)
tensor(0.0533, grad_fn=<NllLossBackward>)
```

**Comment by volcacius** [comment]
I would increase the batch size by a factor of 10, or decrease the learning rate. For simple MNIST examples I suggest you to look at https://github.com/Xilinx/brevitas/tree/master/brevitas_examples/bnn_pynq . There are training scripts with good hyperparameters included.

**Comment by anadb** [comment]
Ok, thank you it worked.
